### PR TITLE
add queryParams when building authority

### LINKED ISSUE #2697: Process the query part of the authority when present
### Microsoft.Identity.Web Library

Microsoft.Identity.Web

### Microsoft.Identity.Web version

2.17.0

### Web app

Sign-in users

### Web API

Protected web APIs (validating tokens)

### Token cache serialization

In-memory caches

### Description

When the authority is provided with a query part, it would be great to keep it and propagate to the MetadataUrl and the MSAL authority

```json
"Authority" : "https://customdomain.com/tenant/v2.0?dc=dcxyw
```

### Reproduction steps

```json
"Authority" : "https://customdomain.com/tenant/v2.0?dc=dcxyw
```

### Error message

_No response_

### Id Web logs

_No response_

### Relevant code snippets

```csharp
"Authority" : "https://customdomain.com/tenant/v2.0?dc=dcxyw
```

### Regression

_No response_

### Expected behavior

?dc=dcxyw is taken into account

### PR DESCRIPTION
**Pull request by kellyyangsong**
Fixes #2697

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: tests/Microsoft.Identity.Web.Test/AuthorityHelpersTests.cs [hunk 1]
```diff
@@ -99,6 +100,72 @@ public void BuildAuthority_CiamAuthorityWithTenant_ReturnsValidAadAuthority()
             Assert.True(preserveAuthority);
         }
 
+        [Fact]
+        public void BuildAuthority_WithQueryParams_ReturnsValidAadAuthority()
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Web.Test/AuthorityHelpersTests.cs:104]: Nice test naming.

**Comment by pmaytak** [comment]
Wouldn't the authority with a query string be an issue here; it'll always fail, since v2.0 would never be at the end.

https://github.com/AzureAD/microsoft-identity-web/blob/0bcdfaf702c0ffd91e244739fccb3631344a6c8b/src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs#L188-L194

https://github.com/AzureAD/microsoft-identity-web/blob/0bcdfaf702c0ffd91e244739fccb3631344a6c8b/src/Microsoft.Identity.Web/AuthorityHelpers.cs#L26-L35

**Comment by kellyyangsong** [comment]
> @kellyyangsong @jennyf19 Wouldn't the authority with a query string be an issue here; it'll always fail, since v2.0 would never be at the end.
> 
> https://github.com/AzureAD/microsoft-identity-web/blob/0bcdfaf702c0ffd91e244739fccb3631344a6c8b/src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs#L188-L194
> 
> https://github.com/AzureAD/microsoft-identity-web/blob/0bcdfaf702c0ffd91e244739fccb3631344a6c8b/src/Microsoft.Identity.Web/AuthorityHelpers.cs#L26-L35

Really good catch, @pmaytak. I can change the logic to ensure that `v2.0` is present and in the right place, if not, add it to the correct spot. I can also write additional tests as our current ones did not catch this.

**Review by pmaytak — DISMISSED** [review]
Also I suggest to test end-to-end to make sure MSAL respects the authority query parameters (I'm not 100% that it does).

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @kellyyangsong

**Review by jennyf19 — APPROVED** [review]

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @kellyyangsong

**Pull request merged**